### PR TITLE
Add delete-entry command to Browser Integration API

### LINKED ISSUE #6887: [Browser Protocol] Request to delete entry in database
## Summary
Whereas third party applications are allowed to access and change a KeePassXC database via KeePassXCs built-in proxy, the protocol does not provide a request to **delete** entries in the database.

Deleting an entry could be secured the same way adding a new group is protected: bring KeePassXC to the foreground and show a pop-up, that asks to confirm deleting the entry.

### PR DESCRIPTION
**Pull request by varjolintu**
[NOTE]: # ( Describe your changes in detail, why is this change required? )
[NOTE]: # ( Explain large or complex code modifications. )
[NOTE]: # ( If it fixes an open issue, please add "Fixes #XXX" )
Adds `delete-entry` command to Browser Integration API. User interaction is needed.

Only entry UUID is accepted as a parameter.
Returns `success` as `true` when entry was deleted, `false` if canceled.

Fixes #6887.

## Testing strategy
[NOTE]: # ( Please describe in detail how you tested your changes. )
[TIP]:  # ( We expect new code to be covered by unit tests and documented with doc blocks! )
Manually.

## Type of change
[NOTE]: # ( Please remove all lines which don't apply. )
- ✅ New feature (change that adds functionality)

### CODE REVIEW

File: src/browser/BrowserAction.cpp [hunk 1]
```diff
@@ -501,6 +504,39 @@ QJsonObject BrowserAction::handleGetTotp(const QJsonObject& json, const QString&
...
+    if (decrypted.isEmpty()) {
+        return getErrorReply(action, ERROR_KEEPASS_CANNOT_DECRYPT_MESSAGE);
+    }
+
+    QString command = decrypted.value("action").toString();
+    if (command.isEmpty() || command.compare("delete-entry", Qt::CaseSensitive) != 0) {
+        return getErrorReply(action, ERROR_KEEPASS_INCORRECT_ACTION);
+    }
+
+    const auto uuid = decrypted.value("uuid").toString();
```
Comment (droidmonkey) [inline src/browser/BrowserAction.cpp:534]: I've been thinking, we really should sanitize the input here. We pass these values directly to some internal function calls without sanitizing them. Should at least determine if it is valid uuid length and character makeup.

Comment (varjolintu) [reply src/browser/BrowserAction.cpp:534]: Got it. I guess there's no existing validation for it?

Comment (droidmonkey) [reply src/browser/BrowserAction.cpp:534]: There might be when you get to Entry::findEntryByUuid but we should catch any anomalies or malicious data well before then.

Comment (varjolintu) [reply src/browser/BrowserAction.cpp:534]: I added some basic checks for UUID to `Tools::isValidUuid()`.

**Comment by purejava** [comment]
I compiled your branch, implemented the `delete-entry` feature in [my Java library](https://github.com/purejava/keepassxc-proxy-access) that accesses KeePassXC via its build-in proxy and tried it out.

It worked like a charm! 🤩

Bildschirmfoto vom 2021-09-08 19-30-26

**Comment by varjolintu** [comment]
> I compiled your branch, implemented the `delete-entry` feature in [my Java library](https://github.com/purejava/keepassxc-proxy-access) that accesses KeePassXC via its build-in proxy and tried it out.
> 
> It worked like a charm! 🤩

I'm still thinking if the dialog should also show the URL just in case. There might be identical titles after all.

**Comment by droidmonkey** [comment]
Can you rebase onto develop?

**Comment by varjolintu** [comment]
> Can you rebase onto develop?

Sure. Done.

**Review by droidmonkey — APPROVED** [review]

**Pull request merged**